File: tests/fixtures/ecosystems/javascript_comprehensive/routes/handlers.js
// SPDX-License-Identifier: MIT
// Copyright (c) 2025-2026 eshu-hq

// #5361 route query-proof matrix: cribs the Express named-handler shape
// proven by internal/parser/engine_javascript_route_handler_test.go.
const express = require("express");
const app = express();

app.get("/catalog/items", listItems);
app.post("/catalog/items", createItem);

function listItems(req, res) {
  res.json([]);
}

function createItem(req, res) {
  res.status(201).json({});
}

File: tests/fixtures/ecosystems/rust_comprehensive/routes/handlers.rs
// SPDX-License-Identifier: MIT
// Copyright (c) 2025-2026 eshu-hq

// #5361 route query-proof matrix: cribs the Axum router shape proven by
// internal/parser/rust_route_entries_test.go
// (TestDefaultEngineParsePathRustEmitsExactFrameworkRouteEntries).
use axum::{routing::get, Router};

fn app() -> Router {
    Router::new().route("/axum/:id", get(axum_show))
}

async fn axum_show() -> &'static str {
    "axum"
}
